src/ray/core_worker/reference_counter.h

```diff
@@ -807,13 +807,15 @@ class ReferenceCounter : public ReferenceCounterInterface,
     /// Callback that will be called when this object
     /// is out of scope or manually freed.
     /// Note: when an object is out of scope, it can still
-    /// have lineage ref count and on_object_ref_delete
+    /// have lineage ref count and the callbacks in object_ref_deleted_callbacks
     /// will be called when lineage ref count is also 0.
     std::vector<std::function<void(const ObjectID &)>>
         on_object_out_of_scope_or_freed_callbacks;
-    /// Callback that will be called when the object ref is deleted
+    /// Callbacks that will be called when the object ref is deleted
     /// from the reference table (all refs including lineage ref count go to 0).
-    std::function<void(const ObjectID &)> on_object_ref_delete;
+    /// Need to store multiple due to message reordering where the callback of the retry
+    /// of the request could be overwritten by the callback of the initial request
```

Comment:
Move the comment to HandleRequest in core worker. In the reference counter context, there is no request